modules/admin-ui/src/main/java/org/opencastproject/adminui/endpoint/UsersEndpoint.java

```diff
@@ -313,6 +315,38 @@ public Response getUsers(@QueryParam("filter") String filter, @QueryParam("sort"
     return Response.ok(gson.toJson(response)).build();
   }
 
+
+  @GET
+  @Path("usersforroles.json")
+  @Produces(MediaType.APPLICATION_JSON)
+  @RestQuery(
+      name = "usersforroles",
+      description = "Returns a list of users",
+      returnDescription = "Returns a JSON representation of the list of user accounts",
+      restParameters = {
+        @RestParameter(name = "roles", isRequired = false, description = "JSON Array", type = STRING),
+      },
+      responses = {
+          @RestResponse(responseCode = SC_OK, description = "The user accounts.")
+      })
+  public Response getsUsersForRoles(@QueryParam("roles") String roles) {
+    List<String> rolesList = gson.fromJson(roles, ArrayList.class);
+    Map<String, Map<String, Object>> roleUserMap = new HashMap<>();
+
+    for (String role : rolesList) {
+      if (!isSanitize()) {
+        User user = userDirectoryService.loadUser(role.replaceFirst(getUserRolePrefix(), ""));
+        if (user != null) {
+          roleUserMap.put(role, generateJsonUser(user));
+        } else {
+          roleUserMap.put(role, null);
+        }
+      }
+    }
+
+    return Response.ok(gson.toJson(roleUserMap)).build();
+  }
```

Comment:
Could this be a security issue? Because I see no security check, but there is a check in "loadUser()" I think. I will test this. Other endpoints in this class are also checking the input and returning a "bad request" in case the input is empty for example.

Reply:
Okay never mind, I did take a second look at this and especially at the other endpoints in this class.